pkg/scheduler/scheduler.go

```diff
@@ -308,7 +308,7 @@ func (s *Scheduler) admit(ctx context.Context, e *entry) error {
 	log.V(2).Info("Workload assumed in the cache")
 
 	s.admissionRoutineWrapper.Run(func() {
-		err := s.client.Update(ctx, newWorkload)
+		err := s.client.Update(ctx, newWorkload.DeepCopy())
```

Comment:
do we need to do something similar on status updates?

Reply:
The status updates that happen in the reconcilers? They don't need to, because the reconcilers get the object from the client, which makes an implicit deep copy. And then we don't reuse those objects.